extensions/opc/src/fetchopc.cpp

```diff
@@ -0,0 +1,239 @@
+  core::Relationship FetchOPCProcessor::Failure("failure", "Retrieved OPC-UA nodes where value cannot be extracted (only if enabled)");
+
+
+  void FetchOPCProcessor::initialize() {
+    //BaseOPCProcessor::initialize();
+
+    // Set the supported properties
+    std::set<core::Property> fetchOPCProperties = {OPCServerEndPoint, NodeID, NodeIDType, NameSpaceIndex};
+    std::set<core::Property> baseOPCProperties = BaseOPCProcessor::getSupportedProperties();
+    fetchOPCProperties.insert(baseOPCProperties.begin(), baseOPCProperties.end());
+    setSupportedProperties(fetchOPCProperties);
+
+    // Set the supported relationships
+    setSupportedRelationships({Success, Failure});
+  }
+
+  void FetchOPCProcessor::onSchedule(const std::shared_ptr<core::ProcessContext> &context, const std::shared_ptr<core::ProcessSessionFactory> &factory) {
+    logger_->log_trace("FetchOPCProcessor::onSchedule");
+
+    translatedNodeIDs_.clear();  // Path might has changed during restart
+
+    BaseOPCProcessor::onSchedule(context, factory);
+
+    if(!configOK_) {
+      return;
+    }
+
+    configOK_ = false;
+
+    std::string value;
+    context->getProperty(NodeID.getName(), nodeID_);
+    context->getProperty(NodeIDType.getName(), value);
+
+    if (value == "String") {
+      idType_ = opc::OPCNodeIDType::String;
+    } else if (value == "Int") {
+      idType_ = opc::OPCNodeIDType::Int;
+    } else if (value == "Path") {
+      idType_ = opc::OPCNodeIDType::Path;
+    } else {
+      // Where have our validators gone?
+      logger_->log_error("%s is not a valid node ID type!", value.c_str());
+    }
+
+    if(idType_ == opc::OPCNodeIDType::Int) {
+      try {
+        int t = std::stoi(nodeID_);
+      } catch(...) {
+        logger_->log_error("%s cannot be used as an int type node ID", nodeID_.c_str());
+        return;
+      }
+    }
+    if(idType_ != opc::OPCNodeIDType::Path) {
+      if(!context->getProperty(NameSpaceIndex.getName(), nameSpaceIdx_)) {
+        logger_->log_error("%s is mandatory in case %s is not Path", NameSpaceIndex.getName().c_str(), NodeIDType.getName().c_str());
+        return;
+      }
+    }
+
+    configOK_ = true;
+  }
+
+  void FetchOPCProcessor::onTrigger(const std::shared_ptr<core::ProcessContext> &context, const std::shared_ptr<core::ProcessSession> &session){
+    if(!configOK_) {
+      logger_->log_error("This processor was not configured properly, yielding. Please check for previous errors in the logs!");
+      yield();
+      return;
+    }
+
+    logger_->log_trace("FetchOPCProcessor::onTrigger");
+
+    std::unique_lock<std::mutex> lock(onTriggerMutex_, std::try_to_lock);
+    if(!lock.owns_lock()){
+      logger_->log_warn("processor was triggered before previous listing finished, configuration should be revised!");
+      return;
+    }
+
+    nodesFound_ = 0;
+    variablesFound_ = 0;
+
+    if(!opc::isConnected(connection_)) {
```

Comment:
The connection code is common between FetchOpc and PutOpc. I suggest moving this to the base class as a function, as `connection_` is already there, and we could avoid the mistake of changing one implementation and forgetting about the other.

Reply:
Added, thanks!